### DISCUSSION TITLE
phpredis cluster setup persistent connection

### DISCUSSION THREAD

**Discussion by exabytes-yllai** [rhubarbgroup/.github#51] [Support (ROC)] [answered]
I am trying to minimize redis latency on redis object cache plugin using phpredis client with persistent connection. 
May I know if the below setup is correct if I wanna establish a persistent connection to my redis cluster?

```php
define('WP_REDIS_CLIENT', 'phpredis');
define('WP_REDIS_IGBINARY',true);
define( 'WP_REDIS_CLUSTER', [
    'tlsv1.2://influasia-serverless-fxwlfg.serverless.apse1.cache.amazonaws.com:6379?alias=node-01&persistent=1',^M
] );
define( 'WP_REDIS_SSL_CONTEXT', [     
    'verify_peer' => false,
]);      
```

Below is my redis.ini (/usr/local/etc/php/conf.d/docker-php-ext-redis.ini) setting. 
```ini
extension=redis
redis.pconnect.pooling_enabled=1
redis.pconnect.connection_limit=10
```

I am getting high latency on my redis statistics with this setup. 
![image](https://github.com/rhubarbgroup/.github/assets/120459363/dbde3308-8ae5-4ac9-ad05-3bde0562d014)

**Comment by tillkruss** [comment, marked as answer]
You might be better off asking in PhpRedis for the `ini` directives.

Using persistent connections will only reduce the initial connection latency, however the majority of commands still need to do round trips. I'd suggest using Relay if latency matters.

**Comment by exabytes-yllai** [comment]
Got it. in that case, can you confirm this is the correct way to establish a persistent connection using redis object cache plugin?
